resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/BasicExecutorFeatureStep.scala

```diff
@@ -260,16 +267,22 @@ private[spark] class BasicExecutorFeatureStep(
         .withUid(pod.getMetadata.getUid)
         .build()
     }
+
+    val policy = kubernetesConf.get(KUBERNETES_ALLOCATION_PODSALLOCATOR) match {
+      case "statefulset" => "Always"
+      case _ => "Never"
+    }
     val executorPodBuilder = new PodBuilder(pod.pod)
       .editOrNewMetadata()
         .withName(name)
         .addToLabels(kubernetesConf.labels.asJava)
+        .addToLabels(SPARK_RESOURCE_PROFILE_ID_LABEL, resourceProfile.id.toString)
```

Comment:
This looks like an orthogonal and good addition. Could you make a separate and simple PR for this?

Reply:
So I actually do use this in this PR, it's so that we can tell the difference between which resource profile a stateful sets pods belong to (I use it in `addToMatchLabels`).

Reply:
Yes, I know. What I mean is that we can merge `.addToLabels(SPARK_RESOURCE_PROFILE_ID_LABEL, resourceProfile.id.toString)` PR independency from this statefulset API.

Reply:
Ah ok, yeah I'll do that :)

Reply:
Dug through the code looks like this is a duplicate change already handled in KubeConf so I'll drop it.

Reply:
Remove it in : https://github.com/apache/spark/pull/34646